components/automate-ui/src/app/entities/automate-settings/automate-settings.requests.ts

```diff
@@ -22,49 +25,111 @@ export class AutomateSettingsRequests {
-      case IngestJobs.MissingNodesForDeletion: {
-        url = `${RETENTION_URL}/nodes/missing-nodes-deletion/config`;
-        break;
+    const body: JobRequestBody = {
+      infra : {
+        job_settings: [
+          {
+            name: 'periodic_purge_timeseries',
+            purge_policies: {
+              elasticsearch: []
+            }
+          }
+        ]
+      },
+      compliance: {
+        job_settings: [
+          {
+            name: 'periodic_purge',
+            purge_policies: {
+            elasticsearch: []
+            }
+          }
+        ]
+      },
+      event_feed: {
+        job_settings: [
+          {
+            name: 'periodic_purge',
+            purge_policies: {
+            elasticsearch: []
+            }
+          }
+        ]
       }
-      case IngestJobs.DeleteNodes: {
-        url = `${RETENTION_URL}/nodes/delete-nodes/config`;
-        break;
+    };
+
+    jobs.forEach((job: IngestJob): void => {
+      let thisJob;
+      let matchedNest;
+
+      switch (job.name) {  // ADD STRONG TYPING
+        case 'delete_nodes':      // fallthrough
+        case 'missing_nodes':     // fallthrough
+        case 'missing_nodes_for_deletion':
+          thisJob = this.unfurlIngestJob(job);
+          body[job.category].job_settings.push(thisJob);
+          break;
+
+        case 'periodic_purge_timeseries': // fallthrough
+        case 'periodic_purge':
+          thisJob = this.unfurlIngestJob(job, true);
+          matchedNest = body[job.category].job_settings.find(item => item.name === job.name);
+          matchedNest.purge_policies.elasticsearch.push(thisJob);
+          break;
+
+        default:
+          break;
       }
-      default:
-        return;
-    }
 
-    const body = {
-      'threshold': job.threshold,
-      'running': job.running
-      // We can also modify how often the job runs (every X time)
-      // but we don't need that now!
-      // 'every': job.every
-    };
+    });
 
-    return this.http.post<any>(url, body);
+    return this.http.put<any>(url, body);
   }
 
+  private unfurlIngestJob(job: IngestJob, nested: boolean = false): UnfurledJob {
+    if (nested) {
+      return {
+        policy_name: job.nested_name,
+        older_than_days: parseInt(job.threshold, 10),
+        disabled: job.disabled
+      };
+    } else {
+      return {
+        name: job.name,
+        threshold: job.threshold,
+        disabled: job.disabled
+      };
+    }
+  }
+
+
   private convertResponseToJobSchedulerStatus(
     respJobSchedulerStatus: RespJobSchedulerStatus): JobSchedulerStatus {
-      const jobs = respJobSchedulerStatus.jobs.map((respJob: RespJob) => new IngestJob(respJob));
-      return new JobSchedulerStatus(respJobSchedulerStatus.running, jobs);
+
+    const allJobs = [];
+
+    Object.keys(respJobSchedulerStatus).forEach((category: string) => {
+      if (respJobSchedulerStatus[category]) {
+        const catJobs = respJobSchedulerStatus[category].jobs
```

Comment:
This restricted type will allow hovering over `.jobs` in the last line to show a real type instead of `any`.
```suggestion
    Object.keys(respJobSchedulerStatus).forEach((category: JobCategories) => {
      if (respJobSchedulerStatus[category]) {
        const catJobs = respJobSchedulerStatus[category].jobs
```